### PR TITLE
Image/VideoSetLabels from glob pattern

### PR DESCRIPTION
**Pull request by tylerganter**
Any instance of `Set` that has an element class that is serializable may be loaded from a numeric or glob element pattern. I also deprecated the class methods in `eta.core.{image, video}`

Example usage:

```
set_labels = etav.VideoSetLabels.from_numeric_patt("%d_tele.json")
set_labels2 = etav.VideoSetLabels.from_glob_patt("*.json")
```

### CODE REVIEW

File: eta/core/image.py [hunk 1]
```diff
@@ -833,6 +834,22 @@ def from_image_labels_patt(cls, image_labels_patt):
             image_set_labels.add(cls._ELE_CLS.from_json(labels_path))
         return image_set_labels
 
+    @classmethod
```
Comment (mattphotonman) [inline eta/core/image.py]: Why not just make this a method of the underlying `Set` class and call it `Set.from_glob_patt()`?  That way we don't have to re-write it for every `Set` subclass.

Comment (tylerganter) [reply eta/core/image.py]: Great point!

File: eta/core/serial.py [hunk 2]
```diff
@@ -824,6 +835,36 @@ def from_dict(cls, d, **kwargs):
         elements = [cls._ELE_CLS.from_dict(dd) for dd in d[cls._ELE_ATTR]]
         return cls(**etau.join_dicts({cls._ELE_ATTR: elements}, kwargs))
 
+    @classmethod
+    def from_element_patt(cls, pattern, pattern_type=NUMERIC):
```
Comment (mattphotonman) [inline eta/core/serial.py]: I suggest adding arguments `*args, **kwargs` that will be passed to the `from_json()` method.  See [here](https://github.com/voxel51/eta/blob/develop/eta/core/serial.py#L388).

Comment (tylerganter) [reply eta/core/serial.py]: happy now?!

File: eta/core/serial.py [hunk 3]
```diff
@@ -49,6 +50,7 @@
 logger = logging.getLogger(__name__)
 
 
+
```
Comment (brimoor) [inline eta/core/serial.py]: unnecessary newline

File: eta/core/serial.py [hunk 4]
```diff
@@ -824,6 +826,47 @@ def from_dict(cls, d, **kwargs):
...
 
+    @classmethod
+    def from_numeric_patt(cls, pattern, *args, **kwargs):
+        '''Creates an instance of `cls` from a numeric pattern of `_ELE_CLS`
+        files.
+
+        Args:
+             pattern: a pattern with one or more numeric sequences
+                example: "/path/to/labels/%05d.json"
+            *args: optional positional arguments for `cls.from_json()`
```
Comment (brimoor) [inline eta/core/serial.py]: `_ELE_CLS.from_json`, not `cls.from_json`

File: eta/core/video.py [hunk 5]
```diff
@@ -1931,10 +1932,9 @@ def from_video_labels_patt(cls, video_labels_patt):
         Returns:
             a `cls` instance
         '''
-        image_set_labels = cls()
-        for labels_path in etau.get_pattern_matches(video_labels_patt):
-            image_set_labels.add(cls._ELE_CLS.from_json(labels_path))
-        return image_set_labels
+        logger.warning("Using deprecated method `from_video_labels_patt`. Use"
+                       " `from_element_numeric_patt` instead.")
+        return cls.from_numeric_patt(video_labels_patt)
```
Comment (brimoor) [inline eta/core/video.py]: `from_element_numeric_patt` is incorrect in this warning message

**Pull request merged**